screenpipe-server/src/db.rs

```diff
@@ -459,34 +484,70 @@ impl DatabaseManager {
     ) -> Result<Vec<SearchResult>, sqlx::Error> {
         let mut results = Vec::new();
 
```

Comment:
@m13v how do you know it works in all use cases and does not break previous behaviour?

Reply:
you mean vector initialization? which use cases would you test?